src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -131,13 +132,7 @@ public void OnComplete(int hr, uint debugTargetCount, VsDebugTargetProcessInfo[]
             {
                 if (_targetProfile != null)
                 {
-                    Assumes.NotNull(_activeProfile);
-#pragma warning disable RS0030 // Do not used banned APIs
-                    ThreadHelper.JoinableTaskFactory.Run(async delegate
-                    {
-                        await _targetProfile.OnAfterLaunchAsync(_launchOptions, _activeProfile);
-                    });
-#pragma warning restore RS0030 // Do not used banned APIs
+                    _threadingService.ExecuteSynchronously(() => _targetProfile.OnAfterLaunchAsync(_launchOptions, _activeProfile));
```

Comment:
Should we always call `OnAfterLaunchAsync()` ? In this implementation we call `OnAfterLaunchAsync()` is only if launch succeeds.